doc/SONiC_OC_NTP_HLD.md

```diff
@@ -0,0 +1,625 @@
++     |  |  +--rw oc-sys-ext:ntp-source-interface?   oc-if:base-interface-ref
++     |  |  +--rw oc-sys-ext:vrf?                    string
+      |  +--ro state
+      |  |  +--ro enabled?                           boolean
+      |  |  +--ro enable-ntp-auth?                   boolean
+      |  |  +--ro auth-mismatch?                     oc-yang:counter64
++     |  |  +--ro oc-sys-ext:ntp-source-interface?   oc-if:base-interface-ref
++     |  |  +--ro oc-sys-ext:vrf?                    string
+      |  +--rw ntp-keys
+      |  |  +--rw ntp-key* [key-id]
+      |  |     +--rw key-id    -> ../config/key-id
+      |  |     +--rw config
+      |  |     |  +--rw key-id?      uint16
+      |  |     |  +--rw key-type?    identityref
+      |  |     |  +--rw key-value?   string
+      |  |     +--ro state
+      |  |        +--ro key-id?      uint16
+      |  |        +--ro key-type?    identityref
+      |  |        +--ro key-value?   string
+      |  +--rw servers
+      |     +--rw server* [address]
++     |        +--rw address    -> ../config/address
++     |        +--rw config
++     |        |  +--rw address?            oc-inet:host
+      |        |  +--rw port?               oc-inet:port-number
+      |        |  +--rw version?            uint8
+      |        |  +--rw association-type?   enumeration
+      |        |  +--rw iburst?             boolean
+      |        |  +--rw prefer?             boolean
++     |        +--ro state
++     |           +--ro address?                 oc-inet:host
+      |           +--ro port?                    oc-inet:port-number
+      |           +--ro version?                 uint8
+      |           +--ro association-type?        enumeration
+      |           +--ro iburst?                  boolean
+      |           +--ro prefer?                  boolean
++     |           +--ro stratum?                 uint8
+      |           +--ro root-delay?              uint32
+      |           +--ro root-dispersion?         uint64
+      |           +--ro offset?                  uint64
++     |           +--ro poll-interval?           uint32
++     |           +--ro oc-sys-ext:peerdelay?    decimal64
++     |           +--ro oc-sys-ext:peeroffset?   decimal64
++     |           +--ro oc-sys-ext:peerjitter?   decimal64
++     |           +--ro oc-sys-ext:selmode?      string
++     |           +--ro oc-sys-ext:refid?        inet:host
++     |           +--ro oc-sys-ext:peertype?     string
++     |           +--ro oc-sys-ext:now?          uint32
++     |           +--ro oc-sys-ext:reach?        uint8
+
+module: sonic-system-ntp
+
++    +--rw sonic-system-ntp
++       +--rw NTP
++       |  +--rw NTP_LIST* [global_key]
++       |     +--rw global_key    enumeration
++       |     +--rw src_intf?     union
++       |     +--rw vrf?          string
++       +--rw NTP_SERVER
++          +--rw NTP_SERVER_LIST* [server_address]
++             +--rw server_address    inet:host
+
+```
+
+### 3.6.2 CLI

```

Comment:
NTPD supports behaving as an NTP server to downstream (downStratum ?) NTP clients. Is there any plan on implementing this? Requests coming in from the field.